### PR TITLE
Prepare for v1.15 development cycle

### PR DESCRIPTION
**Pull request by joestringer**
Depends on: #26237 (which introduces per-branch comment-trigger workflows, so this PR doesn't need them)

This should be merged once we prepare the v1.14 branch.

Differences from the previous dev cycle preparation PR:
- Don't include stable branch CI workflows in the main branch.
- Rely on renovate over dependabot on main branch now.
- Use the `v1.15-dev` version during the dev cycle.

### CODE REVIEW

File: .github/workflows/conformance-clustermesh-v1.14.yaml
Comment (giorio94) [inline .github/workflows/conformance-clustermesh-v1.14.yaml:1]: The conformance-clustermesh workflow on main includes a few additional modifications to test features that were not supported in v1.13 (mainly kvstoremesh and auth modes), as well as to use the CLI helm mode. I think we need to propagate those changes also here.

<details>
  <summary>Diff (excluding the trigger part)</summary>

```diff
diff --git a/.github/workflows/conformance-clustermesh-v1.14.yaml b/.github/workflows/conformance-clustermesh.yaml
--- a/.github/workflows/conformance-clustermesh-v1.14.yaml
+++ b/.github/workflows/conformance-clustermesh.yaml
@@ -66,10 +66,11 @@ concurrency:
 env:
   # renovate: datasource=github-releases depName=kubernetes-sigs/kind
   kind_version: v0.20.0
-  k8s_version: v1.24.12
+  k8s_version: v1.27.1
   # renovate: datasource=github-releases depName=cilium/cilium-cli
   cilium_cli_version: v0.14.8
   cilium_cli_ci_version:
+  CILIUM_CLI_MODE: helm
   clusterName1: cluster1-${{ github.run_id }}
   clusterName2: cluster2-${{ github.run_id }}
   contextName1: kind-cluster1-${{ github.run_id }}
@@ -191,12 +189,18 @@ jobs:
             ipfamily: 'ipv4'
             encryption: 'disabled'
             kube-proxy: 'iptables'
+            kvstoremesh: true
+            cm-auth-mode-1: 'legacy'
+            cm-auth-mode-2: 'legacy'
 
           - name: '2'
             tunnel: 'disabled'
             ipfamily: 'ipv4'
             encryption: 'wireguard'
             kube-proxy: 'none'
+            kvstoremesh: false
+            cm-auth-mode-1: 'migration'
+            cm-auth-mode-2: 'migration'
 
           # IPsec encryption cannot be used with BPF NodePort.
           - name: '3'
@@ -204,6 +208,9 @@ jobs:
             ipfamily: 'ipv4'
             encryption: 'ipsec'
             kube-proxy: 'iptables'
+            kvstoremesh: true
+            cm-auth-mode-1: 'cluster'
+            cm-auth-mode-2: 'cluster'
 
           # IPsec encryption is currently not supported in case of ipv6-only clusters (#23553)
           # Wireguard encryption is currently affected by a bug in case of ipv6-only clusters (#23917)
@@ -212,6 +219,9 @@ jobs:
             ipfamily: 'ipv6'
             encryption: 'disabled'
             kube-proxy: 'none'
+            kvstoremesh: false
+            cm-auth-mode-1: 'legacy'
+            cm-auth-mode-2: 'migration'
 
           # IPsec encryption cannot be used with BPF NodePort.
           - name: '5'
@@ -219,18 +229,27 @@ jobs:
             ipfamily: 'dual'
             encryption: 'ipsec'
             kube-proxy: 'iptables'
+            kvstoremesh: true
+            cm-auth-mode-1: 'migration'
+            cm-auth-mode-2: 'cluster'
 
           - name: '6'
             tunnel: 'vxlan'
             ipfamily: 'ipv4'
             encryption: 'disabled'
             kube-proxy: 'none'
+            kvstoremesh: false
+            cm-auth-mode-1: 'cluster'
+            cm-auth-mode-2: 'cluster'
 
           - name: '7'
             tunnel: 'vxlan'
             ipfamily: 'ipv4'
             encryption: 'wireguard'
             kube-proxy: 'iptables'
+            kvstoremesh: true
+            cm-auth-mode-1: 'cluster'
+            cm-auth-mode-2: 'cluster'
 
           # IPsec encryption cannot be used with BPF NodePort.
           - name: '8'
@@ -238,6 +257,9 @@ jobs:
             ipfamily: 'ipv4'
             encryption: 'ipsec'
             kube-proxy: 'iptables'
+            kvstoremesh: false
+            cm-auth-mode-1: 'cluster'
+            cm-auth-mode-2: 'cluster'
 
         # Tunneling is currently not supported in case of ipv6-only clusters (#17240)
         #  - name: '9'
@@ -245,12 +267,18 @@ jobs:
         #    ipfamily: 'ipv6'
         #    encryption: 'disabled'
         #    kube-proxy: 'none'
+        #    kvstoremesh: true
+        #    cm-auth-mode-1: 'cluster'
+        #    cm-auth-mode-2: 'cluster'
 
           - name: '10'
             tunnel: 'vxlan'
             ipfamily: 'dual'
             encryption: 'wireguard'
             kube-proxy: 'iptables'
+            kvstoremesh: false
+            cm-auth-mode-1: 'cluster'
+            cm-auth-mode-2: 'cluster'
 
     steps:
     - name: Checkout GitHub main
@@ -278,13 +306,22 @@ jobs:
           --helm-set=operator.image.tag=${SHA} \
           --helm-set=operator.image.useDigest=false \
           --helm-set=bpf.masquerade=false \
-          --config monitor-aggregation=none \
-          --rollback=false \
-          --version="
+          --helm-set=bpf.monitorAggregation=none \
+          --helm-set=hubble.enabled=true \
+          --helm-set=hubble.relay.enabled=true \
+          --helm-set=hubble.relay.image.override=quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/hubble-relay-ci:${SHA} \
+          --helm-set=hubble.relay.image.useDigest=false \
+          --helm-set=clustermesh.useAPIServer=true \
+          --helm-set=clustermesh.apiserver.kvstoremesh.enabled=${{ matrix.kvstoremesh }} \
+          --helm-set=clustermesh.apiserver.image.override=quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/clustermesh-apiserver-ci:${SHA} \
+          --helm-set=clustermesh.apiserver.image.useDigest=false \
+          --helm-set=clustermesh.apiserver.kvstoremesh.image.override=quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/kvstoremesh-ci:${SHA} \
+          --helm-set=clustermesh.apiserver.kvstoremesh.image.useDigest=false \
+          "
 
         CILIUM_INSTALL_TUNNEL="--helm-set=tunnel=vxlan"
         if [ "${{ matrix.tunnel }}" == "disabled" ]; then
-          CILIUM_INSTALL_TUNNEL="--helm-set-string=tunnel=disabled \
+          CILIUM_INSTALL_TUNNEL="--helm-set-string=routingMode=native \
             --helm-set=autoDirectNodeRoutes=true \
             --helm-set=ipv4NativeRoutingCIDR=10.240.0.0/12 \
             --helm-set=ipv6NativeRoutingCIDR=fd00:10:240::/44"
@@ -317,18 +354,12 @@ jobs:
             ;;
         esac
 
-        CILIUM_INSTALL_L7_PROXY="--helm-set=l7Proxy=true"
-        if [ "${{ matrix.encryption }}" == "wireguard" ]; then
-          # Wireguard (--enable-wireguard) is not compatible with L7 proxy (--enable-l7-proxy)
-          CILIUM_INSTALL_L7_PROXY="--helm-set=l7Proxy=false"
+        CILIUM_INSTALL_ENCRYPTION=""
+        if [ "${{ matrix.encryption }}" != "disabled" ]; then
+          CILIUM_INSTALL_ENCRYPTION="--helm-set=encryption.enabled=true \
+            --helm-set=encryption.type=${{ matrix.encryption }}"
         fi
 
-        HUBBLE_ENABLE_DEFAULTS="--chart-directory=install/kubernetes/cilium \
-          --helm-set=hubble.relay.image.override=quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/hubble-relay-ci:${SHA} \
-          --helm-set=hubble.relay.image.useDigest=false"
-        CLUSTERMESH_ENABLE_DEFAULTS="--apiserver-image=quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/clustermesh-apiserver-ci \
-          --apiserver-version=${SHA} --service-type=NodePort"
-
         CONNECTIVITY_TEST_DEFAULTS="--hubble=false \
           --flow-validation=disabled \
           --multi-cluster=${{ env.contextName2 }} \
@@ -343,10 +374,9 @@ jobs:
             --test='!/pod-to-cidr'"
         fi
 
-        echo cilium_install_defaults="${CILIUM_INSTALL_DEFAULTS} ${CILIUM_INSTALL_TUNNEL} ${CILIUM_INSTALL_IPFAMILY} ${CILIUM_INSTALL_L7_PROXY}" >> $GITHUB_OUTPUT
-        echo hubble_enable_defaults=${HUBBLE_ENABLE_DEFAULTS} >> $GITHUB_OUTPUT
+        echo cilium_install_defaults="${CILIUM_INSTALL_DEFAULTS} ${CILIUM_INSTALL_TUNNEL} \
+          ${CILIUM_INSTALL_IPFAMILY} ${CILIUM_INSTALL_ENCRYPTION}" >> $GITHUB_OUTPUT
         echo connectivity_test_defaults=${CONNECTIVITY_TEST_DEFAULTS} >> $GITHUB_OUTPUT
-        echo clustermesh_enable_defaults=${CLUSTERMESH_ENABLE_DEFAULTS} >> $GITHUB_OUTPUT
 
         echo kind_pod_cidr_1=${KIND_POD_CIDR_1} >> $GITHUB_OUTPUT
         echo kind_svc_cidr_1=${KIND_SVC_CIDR_1} >> $GITHUB_OUTPUT
@@ -432,44 +462,39 @@ jobs:
         ref: ${{ needs.setup-report.outputs.sha }}
         persist-credentials: false
 
+    - name: Create the IPSec secret in both clusters
+      if: matrix.encryption == 'ipsec'
+      run: |
+        SECRET="3 rfc4106(gcm(aes)) $(openssl rand -hex 20) 128"
+        kubectl --context ${{ env.contextName1 }} create -n kube-system secret generic cilium-ipsec-keys --from-literal=keys="${SECRET}"
+        kubectl --context ${{ env.contextName2 }} create -n kube-system secret generic cilium-ipsec-keys --from-literal=keys="${SECRET}"
+
     - name: Install Cilium in cluster1
       run: |
-        # Using the deprecated flag --cluster-name due to cilium/cilium-cli#1347
-        # --helm-set cluster.name ${{ env.clusterName1 }}
+        # Explicitly configure the NodePort to make sure that it is different in
+        # each cluster, to workaround #24692
         cilium --context ${{ env.contextName1 }} install \
           ${{ steps.vars.outputs.cilium_install_defaults }} \
-          --encryption ${{ matrix.encryption }} \
-          --cluster-name ${{ env.clusterName1 }} \
-          --helm-set cluster.id=1
+          --helm-set cluster.name=${{ env.clusterName1 }} \
+          --helm-set cluster.id=1 \
+          --helm-set clustermesh.apiserver.service.nodePort=32379 \
+          --helm-set clustermesh.apiserver.tls.authMode=${{ matrix.cm-auth-mode-1 }}
 
-    - name: Copy the IPsec secret to cluster2, as they must match
-      if: matrix.encryption == 'ipsec'
+    - name: Copy the Cilium CA secret to cluster2, as they must match
       run: |
-        kubectl --context ${{ env.contextName1 }} get secret -n kube-system cilium-ipsec-keys -o yaml |
+        kubectl --context ${{ env.contextName1 }} get secret -n kube-system cilium-ca -o yaml |
           kubectl --context ${{ env.contextName2 }} create -f -
 
     - name: Install Cilium in cluster2
       run: |
-        # Using the deprecated form --cluster-name due to cilium/cilium-cli#1347
-        # --helm-set cluster.name ${{ env.clusterName2 }}
+        # Explicitly configure the NodePort to make sure that it is different in
+        # each cluster, to workaround #24692
         cilium --context ${{ env.contextName2 }} install \
           ${{ steps.vars.outputs.cilium_install_defaults }} \
-          --encryption ${{ matrix.encryption }} \
-          --cluster-name ${{ env.clusterName2 }} \
+          --helm-set cluster.name=${{ env.clusterName2 }} \
           --helm-set cluster.id=255 \
-          --inherit-ca ${{ env.contextName1 }}
-
-    - name: Enable Hubble
-      run: |
-        cilium --context ${{ env.contextName1 }} hubble enable ${{ steps.vars.outputs.hubble_enable_defaults }} --wait=false
-        cilium --context ${{ env.contextName2 }} hubble enable ${{ steps.vars.outputs.hubble_enable_defaults }} --wait=false --relay=false
-        cilium --context ${{ env.contextName1 }} status --wait
-        cilium --context ${{ env.contextName2 }} status --wait
-
-    - name: Enable Cluster Mesh
-      run: |
-        cilium --context ${{ env.contextName1 }} clustermesh enable ${{ steps.vars.outputs.clustermesh_enable_defaults }}
-        cilium --context ${{ env.contextName2 }} clustermesh enable ${{ steps.vars.outputs.clustermesh_enable_defaults }}
+          --helm-set clustermesh.apiserver.service.nodePort=32380 \
+          --helm-set clustermesh.apiserver.tls.authMode=${{ matrix.cm-auth-mode-2 }}
 
     - name: Wait for cluster mesh status to be ready
       run: |
@@ -515,7 +540,7 @@ jobs:
 
         kubectl config use-context ${{ env.contextName1 }}
         kubectl get pods --all-namespaces -o wide
-        cilium sysdump --output-filename cilium-sysdump-context1-${{ join(matrix.*, '-') }}
+        cilium sysdump --output-filename cilium-sysdump-context1-final-${{ join(matrix.*, '-') }}
 
         kubectl config use-context ${{ env.contextName2 }}
         kubectl get pods --all-namespaces -o wide
```
</details>

Comment (joestringer) [reply .github/workflows/conformance-clustermesh-v1.14.yaml:1]: Thanks for the diff! I'll try to find a moment to pull this in.

**Review by squeed — APPROVED** [review]
reviewed for my files.

**Review by bimmlerd — APPROVED** [review]
contributing changes LGTM

**Review by jschwinger233 — APPROVED** [review]
Datapath looks good, thanks!

**Review by nbusseneau — CHANGES_REQUESTED** [review]
Discussed with Joe offline:

- Since we are close to merging #26237, we will merge it first before branching off as it will simplify the process considerably.
- In particular, for this PR it means we do not need to make copies of the workflows for 1.14. Instead, workflows for 1.14 will live directly on the 1.14 branch. This is handled as part of #26625.

File: .github/workflows/conformance-clustermesh-v1.14.yaml
Comment (joestringer) [reply .github/workflows/conformance-clustermesh-v1.14.yaml:1]: I see that this has been applied to the corresponding workflow on the v1.14 branch, so no need to apply it here now (thanks to Ariane!)

File: .github/workflows/ci-images-garbage-collect.yaml [hunk 1]
```diff
@@ -19,6 +19,6 @@ jobs:
       - uses: docker://quay.io/cilium/scruffy:v0.0.3@sha256:ca997451b739cbf03c204cb2523a671c31c61edc606aa5d20dc3560bc7f25bc7
         with:
           entrypoint: scruffy
-          args: --git-repository=./ --stable-branches=origin/main,origin/v1.9,origin/v1.10,origin/v1.11,origin/v1.12,origin/v1.13
+          args: --git-repository=./ --stable-branches=origin/main,origin/v1.9,origin/v1.10,origin/v1.11,origin/v1.12,origin/v1.13,origin/v1.14
```
Comment (nbusseneau) [inline .github/workflows/ci-images-garbage-collect.yaml:22]: [Nit] Should we remove v1.9 and v1.10 from here?

File: .github/workflows/loki.yaml [hunk 2]
```diff
@@ -14,26 +14,31 @@ on:
       - Cilium L4LB XDP (ci-l4lb-1.11)
       - Cilium L4LB XDP (ci-l4lb-1.12)
       - Cilium L4LB XDP (ci-l4lb-1.13)
+      - Cilium L4LB XDP (ci-l4lb-1.14)
```
Comment (nbusseneau) [inline .github/workflows/loki.yaml:17]: [Useless nit] The `ci-xxx-1.xx` workflows do not exist anymore, now it's only the regular non `-1.xx` versions for all branches. Not that it matters considering Loki is disabled anyway...

**Review by nbusseneau — APPROVED** [review]

**Review by michi-covalent — APPROVED** [review]
😇

**Review by squeed — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by viktor-kurchenko — APPROVED** [review]

File: .github/renovate.json5 [hunk 3]
```diff
@@ -191,6 +193,7 @@
       "allowedVersions": "22.04",
       "matchBaseBranches": [
         "main",
+        "v1.14",
```
Comment (tklauser) [inline .github/renovate.json5:196]: Can't comment on the actual line, but I think we'll need to add a section to restrict Go toolchain to the currently used 1.20 in `1.14` and avoid upgrading to 1.21 once it's out:

```json
    {
      "matchPackageNames": [
        "docker.io/library/golang",
        "go"
      ],
      "allowedVersions": "<1.21",
      "matchBaseBranches": [
        "v1.14"
      ]
    },
```

**Review by tklauser — CHANGES_REQUESTED** [review]

**Review by nathanjsweet — APPROVED** [review]

File: .github/workflows/ci-images-garbage-collect.yaml (hunk 1 above)
Comment (joestringer) [reply .github/workflows/ci-images-garbage-collect.yaml:22]: Yeah we probably should. I have another PR in the works to drop all 1.11 or earlier references, which should wait until the release. I can add this change into that one.

File: .github/workflows/loki.yaml (hunk 2 above)
Comment (joestringer) [reply .github/workflows/loki.yaml:17]: Yeah I was debating if we should just be deleting the loki workflow instead.

File: .github/renovate.json5 (hunk 3 above)
Comment (joestringer) [reply .github/renovate.json5:196]: Added.

Comment (joestringer) [reply .github/renovate.json5:196]: I've also added similar logic for the alpine image bumps.

File: .github/workflows/loki.yaml (hunk 2 above)
Comment (joestringer) [reply .github/workflows/loki.yaml:17]: https://github.com/cilium/cilium/pull/26676

File: .github/workflows/ci-images-garbage-collect.yaml (hunk 1 above)
Comment (joestringer) [reply .github/workflows/ci-images-garbage-collect.yaml:22]: https://github.com/cilium/cilium/pull/26681

**Review by tklauser — APPROVED** [review]

**Review by aanm — APPROVED** [review]

**Pull request merged**